src/component/LayerCarousel/LayerCarousel.tsx

```diff
@@ -273,28 +287,32 @@ export default class LayerCarousel extends React.Component<LayerCarouselProps, L
 
     const extent = map.getView().calculateExtent();
     const mapProjection = map.getView().getProjection().getCode();
-    return this.props.layers.map((layer: any) => {
+    return this.props.layers.map((layer) => {
       const staticImageUrl = layer.get('staticImageUrl');
       const previewImageRequestUrl = layer.get('previewImageRequestUrl');
       let layerFt, requestUrl;
       if (!staticImageUrl) {
         if (layer.get('type') === 'WMTS' && previewImageRequestUrl) {
           requestUrl = previewImageRequestUrl;
-          layerFt = layer.getSource().getLayer();
+          layerFt = (layer.getSource() as OlSourceWMTS).getLayer();
         } else {
-          const source = layer.getSource();
+          const source = layer.getSource() as OlSourceTileWMS;
           const params = source.getParams();
           const layersKey = Object.keys(params).find(p => p.toLowerCase() === 'layers');
           layerFt = source.getParams()[layersKey];
           if (source.getUrls) {
             requestUrl = source.getUrls()[0];
           } else {
+            // TODO Check why we need to get the source of a source?
+            // @ts-ignore
             requestUrl = source.getSource().getUrl();
```

Comment:
I think this should be `source.getUrl()` as layer is a `WMSLayer` the source could be either a `TileWMS` or a `ImageWMS`. The former having `getUrls` and the later `getUrl`.

The check could be improved by using instanceof instead of the explicit type cast to `TileWMS`

Reply:
Yeah, you are right. But somehow I doubt that cluster sources are supported when normal vector sources aren't ...

Sadly the component does not specify which kind of layers it accepts as far as I see.

```
          if (source instanceof TileWMS) {
            requestUrl = source.getUrls()[0];
          } else if (source instanceof ImageWMS) {
            requestUrl = source.getUrl();
          }
```

or


```
          if (source instanceof TileWMS) {
            requestUrl = source.getUrls()[0];
          } else if (source instanceof ImageWMS || source instanceof Vector) {
            requestUrl = source.getUrl();
          } else if (source instanceof Cluster) {
            requestUrl = source.getSource().getUrl();
          }
```

Reply:
Just noticed the call to `getParams` which only exists on `TileWMS` or `ImageWMS`.

Reply:
i doubt that the code is used to be honest. Just wanted to show a possible way where it partly makes sense. but in most cases, it should be layer.getSource() of course... So just ignore my ocmment...